src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -68,19 +67,20 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<YoutubePost> newPost;
     try {
-      List<YoutubePost> newPost = YoutubeApi.getYoutubePost();
+      newPost = YoutubeApi.getYoutubePost();
     } catch (YoutubeApiException e) {
       System.out.println("Error: Youtube api returning exception" + e);
       response.sendError(500, "An error occurred while fetching Youtube Posts");
       return;
     }
     if (threadTitles.size() <= 0) {
       for (int i = 0; i < 5; i++) {
```

Comment:
Can you turn this to a for each loop just to make sure we get all of the posts?

Reply:
+1, great suggestion. It's fine if we just have one post for now, we will retrieve multiple posts in a follow up

Reply:
The getYoutubePost() method returns a list of all the youtube posts

Reply:
But can you iterate over the list of posts instead?

``` java
for (YoutubePost post : newPosts) {
  // Do something
}
```

Reply:
ok, got it!